### PR TITLE
Adds PyProject toml

### PR DESCRIPTION
**Pull request by atruslow**
The docs on this are pretty lacking, but I followed along with [this](https://snarky.ca/what-the-heck-is-pyproject-toml/)

homer

### CODE REVIEW

**Comment by atruslow** [comment]
@chrisantonellis 

`bdist_wheel` works locally, in the container. is it something with the github action? 

```
root@7686ffbabd5e:/app# python setup.py bdist_wheel  --verbose
running bdist_wheel
running build
installing to build/bdist.linux-x86_64/wheel
running install
running install_egg_info
running egg_info
writing little_cheesemonger.egg-info/PKG-INFO
writing dependency_links to little_cheesemonger.egg-info/dependency_links.txt
writing entry points to little_cheesemonger.egg-info/entry_points.txt
writing requirements to little_cheesemonger.egg-info/requires.txt
writing top-level names to little_cheesemonger.egg-info/top_level.txt
reading manifest file 'little_cheesemonger.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'little_cheesemonger.egg-info/SOURCES.txt'
Copying little_cheesemonger.egg-info to build/bdist.linux-x86_64/wheel/little_cheesemonger-0.1.0-py3.8.egg-info
running install_scripts
adding license file "LICENSE" (matched pattern "LICEN[CS]E*")
creating build/bdist.linux-x86_64/wheel/little_cheesemonger-0.1.0.dist-info/WHEEL
creating 'dist/little_cheesemonger-0.1.0-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'little_cheesemonger-0.1.0.dist-info/LICENSE'
adding 'little_cheesemonger-0.1.0.dist-info/METADATA'
adding 'little_cheesemonger-0.1.0.dist-info/WHEEL'
adding 'little_cheesemonger-0.1.0.dist-info/entry_points.txt'
adding 'little_cheesemonger-0.1.0.dist-info/top_level.txt'
adding 'little_cheesemonger-0.1.0.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
root@7686ffbabd5e:/app# ls dist/
little_cheesemonger-0.1.0-py3-none-any.whl

```

**Comment by atruslow** [comment]
I even ran the verify code 

```
root@7686ffbabd5e:/app# python .github/actions/verify-wheel/test_module_import.py little_cheesemonger
Imported little_cheesemonger version 0.1.0

```

**Comment by atruslow** [comment]
I fixed the failing action, I was missing a `packages = find:` line

File: setup.py [hunk 1]
```diff
@@ -1,44 +1,7 @@
...
-import setuptools
-
-_version_re = re.compile(r"__version__\s+=\s+(.*)")
-with open("little_cheesemonger/__init__.py", "rb") as f:
-    _match = _version_re.search(f.read().decode("utf-8"))
-    if _match is None:
-        print("No version found")
-        raise SystemExit(1)
-    version = str(ast.literal_eval(_match.group(1)))
+#!/usr/bin/env python
```
Comment (chrisantonellis) [inline setup.py:2]: could you add a comment saying that this is required for installing in editable mode?

Comment (atruslow) [reply setup.py:1]: yep! added

File: docker-compose.yaml [hunk 2]
```diff
@@ -7,8 +7,9 @@ services:
       dockerfile: ./docker/Dockerfile
       context: .
     image: little-cheesemonger_devbox
+    command: /bin/bash
     volumes:
-      - ./little_cheesemonger:/app/little_cheesemonger
```
Comment (chrisantonellis) [inline docker-compose.yaml:11]: why was this change needed? my original intention was to only mount the files that needed to update for the "editable package" part to work, i have the files mounted to the container to a minimum essentially. requirements files are copied in the dockerfile etc

**Review by chrisantonellis — APPROVED** [review]

**Comment by chrisantonellis** [comment]
thanks for the contribution!!
